test/stripe/subscriptions/subscription_test.exs

```diff
@@ -33,6 +33,7 @@ defmodule Stripe.SubscriptionTest do
 
     test "delete_discount/2 deletes a subscription's discount" do
       {:ok, subscription} = Stripe.Subscription.retrieve("sub_123")
-      assert {:ok, %Stripe.Subscription{discount: nil}} = Stripe.Subscription.delete_discount(subscription)
+      assert {:ok, coupon} = Stripe.Coupon.retrieve("35OFF")
+      assert {:ok, coupon} = Stripe.Subscription.delete_discount(subscription)
```

Comment:
We might want to think about this one.
`delete_discount` is a utility function from the Ruby library, it's not on the Subscription endpoint in the core API. It is a nice utility function, and IMO should return the subscription with no discount—as the previous test which was there has it. Alternatively, we can drop the whole function and force users to use `Stripe.Discount.delete/1`.

Reply:
There's a utility endpoint here, too: https://stripe.com/docs/api/curl#delete_subscription_discount

Reply:
Also note that this is broken because `stripe-mock` appears to just be sending back a nested coupon hash.